python/ml4ir/base/model/scoring/Monte_Carlo_Scorer.py

```diff
@@ -0,0 +1,131 @@
+            and the loss function for the model
+        file_io : `FileIO` object
+            FileIO object that handles read and write
+        aux_loss : `RelevanceLossBase` object
+            Auxiliary loss to be used in conjunction with the primary loss
+        aux_loss_weight: float
+            Floating point number in [0, 1] to indicate the proportion of the auxiliary loss
+            in the total final loss value computed using a linear combination
+            total loss = (1 - aux_loss_weight) * loss + aux_loss_weight * aux_loss
+        aux_metrics: List of keras.metrics.Metric
+            Keras metric list to be computed on the aux label
+        output_name : str, optional
+            Name of the output that captures the score computed by the model
+        logger : Logger, optional
+            Logging handler
+        logs_dir : str, optional
+            Path to the logging directory
+
+        Notes
+        -----
+        logs_dir : Used to point model architectures to local logging directory,
+            primarily for saving visualizations.
+        """
+        super().__init__(feature_config=feature_config,
+                         model_config=model_config,
+                         interaction_model=interaction_model,
+                         loss=loss,
+                         aux_loss=aux_loss,
+                         aux_loss_weight=aux_loss_weight,
+                         aux_metrics=aux_metrics,
+                         output_name=output_name,
+                         logger=logger,
+                         file_io=file_io,
+                         logs_dir=logs_dir,
+                         **kwargs)
+        self.monte_carlo_inference_trials = self.model_config["monte_carlo_inference_trials"]["value"]
+
+    def call(self, inputs: Dict[str, tf.Tensor], training=None):
+        """
+        Compute score from input features
+
+        Parameters
+        --------
+        inputs : dict of tensors
+            Dictionary of input feature tensors
+
+        Returns
+        -------
+        scores : dict of tensor object
+            Tensor object of the score computed by the model
+        """
+        scores = self.make_pred(inputs, training=False)
+        for _ in range(self.monte_carlo_inference_trials):
+            scores += self.make_pred(inputs, training=True)
+        scores = tf.divide(scores, tf.constant(self.monte_carlo_inference_trials + 1, dtype=tf.float32))
+        return {self.output_name: scores}
+
+    def make_pred(self, inputs: Dict[str, tf.Tensor], training=None):
+        # Apply feature layer and transform inputs
+        features = self.interaction_model(inputs, training=training)
+
+        # Apply architecture op on train_features
+        features[FeatureTypeKey.LOGITS] = self.architecture_op(features, training=training)
+
+        # Apply final activation layer
+        scores = self.loss_op.final_activation_op(features, training=training)
+
+        return scores
```

Comment:
@mohazahran I believe I had a comment on replacing `make_pred` with `super().call(...)` as it is a duplicate. Did you get a chance to try?